src/build/mod.rs

```diff
@@ -547,12 +545,56 @@ impl Internals {
         };
         trace!("Specified work: {:?}", work);
 
-        match work {
+        let cargo_build = if let WorkStatus::NeedsCargo(..) = work {
+            true
+        } else {
+            false
+        };
+
+        let result = match work {
             // Cargo performs the full build and returns
             // appropriate diagnostics/analysis data
             WorkStatus::NeedsCargo(package_arg) => cargo::cargo(self, package_arg, progress_sender),
             WorkStatus::Execute(job_queue) => job_queue.execute(self, progress_sender),
+        };
+
+        if let BuildResult::Success(.., true) = result {
+            let elapsed = start.elapsed();
+            *self.last_build_duration.write().unwrap() = Some(elapsed);
+            info!(
+                "{} finished in {:.1?}",
+                if cargo_build { "cargo" } else { "rustc" },
```

Comment:
Could we not include this here, please? I don't think it's important to disambiguate between the two and if someone needs more specific info about what kind of work is being done then there's a lot of supporting debug info for what kind of build is being performed.

Reply:
I added the info log as I think it's useful to have the total build time more easily available to the users, as now can be viewed with `rls::build=info` without other potentially noisy output. This is similar to cargo output. 
```
  Finished dev [unoptimized + debuginfo] target(s) in 1m 56s
```
So it seemed nice to have, though I guess I could add it client-side using the progress messages.

I added cargo/rustc as otherwise it would just say "build finished in ..." so why not add more build-time affecting information in the same length?

In any case I'm happy to remove it, or just the cargo/rustc bit. Which are you after?

Reply:
Don't get me wrong, I really like the `info` and `build finished in ..` info since it's really, well, informative :sweat_smile:
It's just that I think cargo/rustc differentiation shouldn't be included since that's more of an optimization/internal detail; if the user wants to know the difference they probably want a lot more debug logs anyways, which then contains a lot more info about the build and its type.

Reply:
Ok np